microsoft/testsuites/network/networksettings.py

```diff
@@ -264,3 +264,59 @@ def validate_device_gro_lro_settings_change(self, node: Node, log: Logger) -> No
                 "GRO and LRO settings for all the devices are fixed and cannot be"
                 " changed. Skipping test."
             )
+
+    @TestCaseMetadata(
+        description="""
+            This test case verifies changing device's RSS hash key takes
+            into affect.
+
+            Steps:
+            1. Skip the test if the kernel version is any less than LTS 5.
+            2. Get all the device's RSS hash key values.
+            3. Swap the last 2 characters of original hash key to make a new hash key.
+            4. Validate changing the hash key setting using the new hash key.
+            5. Revert back the settings to original values.
+        """,
+        priority=2,
+    )
+    def validate_device_rss_hash_key_change(self, node: Node, log: Logger) -> None:
+        if node.os.information.version.compare("5.0.0") < 0:
+            raise SkippedException(
+                f"The kernel version {node.os.information.version} does not support"
+                " changing RSS hash key."
+            )
+
+        ethtool = node.tools[Ethtool]
+        try:
+            devices_rss_hkey_info = ethtool.get_all_device_rss_hash_key()
+        except UnsupportedOperationException as identifier:
+            raise SkippedException(identifier)
+
+        for device_hkey_info in devices_rss_hkey_info:
+            original_hkey = device_hkey_info.rss_hash_key
+            # Swap the last 2 characters of the original hash key to make new hash key.
+            split_hkey = original_hkey.rsplit(":", 1)
+            swapped_part = "".join(
+                [
+                    split_hkey[1][x : x + 2][::-1]
+                    for x in range(0, len(split_hkey[1]), 2)
+                ]
+            )
+
+            new_hkey = split_hkey[0] + ":" + swapped_part
```

Comment:
Use f string to format it. `f"{split_hkey[0]}:{swapped_part}"`

Reply:
call the variable `expected_key`. It's clearer in test case, expected v.s. actual.